N2A/src/gov/sandia/n2a/ui/settings/SettingsRepo.java

```diff
@@ -785,38 +769,52 @@ public void rebuild ()
     {
         if (! needRebuild) return;
 
-        List<MNode> modelContainers     = new ArrayList<MNode> ();
-        List<MNode> referenceContainers = new ArrayList<MNode> ();
+        Map<String, List<MNode>> existingContainers = new HashMap<String, List<MNode>> ();
         String primary = AppData.state.get ("Repos", "primary");
         for (MNode repo : repoModel.repos)
         {
             String repoName = repo.key ();
             boolean isPrimary = repoName.equals (primary);
             if (! repo.getBoolean ("visible")  &&  ! repo.getBoolean ("editable")  &&  ! isPrimary) continue;
 
-            MNode models     = getModels     (repoName);
-            MNode references = getReferences (repoName);
+            Map<String, MNode> dirs = new HashMap<String, MNode> ();
+            existing.forEach ((k,v) -> dirs.put(k, getExisting(k, repoName)));
             if (isPrimary)
             {
-                modelContainers    .add (0, models);
-                referenceContainers.add (0, references);
+                dirs.forEach ((k,v) -> {
+                    List<MNode> nodes = existingContainers.get (k);
+                    if(nodes == null) 
+                    {
+                        existingContainers.put (k, new ArrayList<MNode> ());
+                        nodes = existingContainers.get(k);
+                    }
+                    nodes.add (0, v);
+                });
```

Comment:
The rebuild process is similar to the static init in AppData. Below the loop at line 774 there should be an inner loop that iterates over actual dirs in the repo, then does the same action as the original code, adding the MDir to either the front or back of the appropriate container collection.

Reply:
I have restructured that code to be more inline with what you said here. Would you mind giving that section another look over to ensure that the logic is correct?